src/main/resources/db/migration/V1_186__update_referral_performance_report_number_of_attendances_srf.sql

```diff
@@ -0,0 +1,19 @@
+create or replace function performance_report_attendances_count(referral_id UUID)
+  returns integer
+as
+$body$
+SELECT count(*)
+FROM appointment a
+WHERE a.referral_id = $1
+	AND a.stale = false
+	AND (a.superseded is null OR a.superseded = false)
+	AND ((a.did_session_happen is null AND (a.late = true OR a.attended = 'YES'))
+	OR (a.did_session_happen = true AND a.attended = 'YES'))
+	AND NOT EXISTS
+    (
+        SELECT 1
```

Comment:
are we selecting only delivery session appointment here. If that is the case we could have did a join in delivery session appointments table instead of checking not exists

Reply:
From what I can tell not exists is more efficient than a join (for a negative case), but we can benchmark time taken once we're in preprod. @kth13

Reply:
Ok sure Andrew.